cadence/contracts/flix_registry.cdc

```diff
@@ -0,0 +1,172 @@
+        fun lookup(idOrAlias: String): FLIX?
+
+        access(all)
+        fun getIds(): [String]
+
+        access(all)
+        fun getAllAlias(): {String: String}
+    }
+
+    access(all)
+    resource interface Admin {
+
+        /// Add the flix to the registry and add or update the alias to point to this flix
+        access(all) 
+        fun publish(alias: String, flix: FLIX)
+
+        access(all)
+        fun link(alias: String, id: String)
+
+        access(all)
+        fun unlink(alias: String)
+
+        access(all)
+        fun deprecate(idOrAlias: String)
+    }
+
+    access(all)
+    resource interface Removable {
+        access(all)
+        fun remove(id: String): FLIX?
+    }
+
+    access(all)
+    resource Registry: Queryable, Removable, Admin {
+
+        access(account) var flixes: {String: FLIX}
+        access(account) var aliases: {String: String}
+        access(account) var cadenceBodyHashes: {String: FLIX}
+
+        access(all) 
+        fun publish(alias: String, flix: FLIX) {
+            self.flixes[flix.id] = flix
+            self.aliases[alias] = flix.id
+            emit Published(alias: alias, id: flix.id, cadenceBodyHash: flix.cadenceBodyHash)
+        }
+
+        access(all)
+        fun link(alias: String, id: String) {
+            self.aliases[alias] = id
+            emit AliasLinked(alias: alias, id: id)
+        }
+
+        access(all)
+        fun unlink(alias: String) {
+            self.aliases.remove(key: alias)
+            emit AliasUnlinked(alias: alias)
+        }
+
+        access(all)
+        fun lookup(idOrAlias: String): FLIX? {
+            if self.aliases.containsKey(idOrAlias) {
+                return self.flixes[self.aliases[idOrAlias]!]
+            }
+            return self.flixes[idOrAlias]
+        }
+
+        access(all)
+        fun resolve(cadenceBodyHash: String): FLIX? {
+            return self.cadenceBodyHashes[cadenceBodyHash]
+        }
+
+        access(all)
+        fun deprecate(idOrAlias: String) {
+            var flix = self.lookup(idOrAlias: idOrAlias) ?? panic("FLIX does not exist with the given id or alias")
+            flix.status = FLIXStatus.deprecated
+            self.flixes[flix.id] = flix
+            emit Deprecated(id: flix.id)
+        }
+
+        access(all)
+        fun remove(id: String): FLIX? {
+            let removed = self.flixes.remove(key: id)
+            emit Removed(id: id)
+            return removed
+        }
+
+        access(all)
+        fun getIds(): [String] {
+            return self.flixes.keys
+        }
+
+        access(all)
+        fun getAllAlias(): {String: String} {
+            return self.aliases
+        }
+
+        init() {
+            self.flixes = {}
+            self.aliases = {}
+            self.cadenceBodyHashes = {}
+        }
+    }
+
+    access(all)
+    fun createRegistry(): @Registry {
+        return <- create Registry()
+    }
+
+    init() {
+        emit ContractInitialized()
```

Comment:
here we should have some standard paths that we can use in the contract like the storagePath of the stable registry and the public link.

Reply:
for now i created a utility function to create Storage and Public path based on registry name. if we need some other solution, i'm happy to discuss
```
    access(all)
    fun PublicPath(name: String): PublicPath {
        return PublicPath(identifier: "flix_".concat(name))!
    }

        access(all)
    fun StoragePath(name: String): StoragePath {
        return StoragePath(identifier: "flix_".concat(name))!
    }
```